Purchases/Public/Purchases.swift

```diff
@@ -896,54 +898,38 @@ public extension Purchases {
         purchasesOrchestrator.paymentDiscount(forProductDiscount: discount, product: product, completion: completion)
     }
 
-    // swiftlint:disable line_length
-    @available(iOS 12.0, macOS 10.14, macCatalyst 13.0, tvOS 12.0, watchOS 6.2, *)
-    private func modernEligibilityHandler(maybeReceiptData data: Data,
-                                          productIdentifiers: [String],
-                                          completion receiveEligibility: @escaping ([String: IntroEligibility]) -> Void) {
-        introEligibilityCalculator
-            .checkTrialOrIntroductoryPriceEligibility(with: data,
-                                                      productIdentifiers: Set(productIdentifiers)) { receivedEligibility, maybeError in
-                if let error = maybeError {
-                    Logger.error(Strings.receipt.parse_receipt_locally_error(error: error))
-                    self.backend.getIntroEligibility(appUserID: self.appUserID,
-                                                     receiptData: data,
-                                                     productIdentifiers: productIdentifiers) { result, maybeAnotherError in
-                        if let intoEligibilityError = maybeAnotherError {
-                            Logger.error(String(format: "Unable to getIntroEligibilityForAppUserID: %@",
-                                                intoEligibilityError.localizedDescription))
-                        }
-                        self.operationDispatcher.dispatchOnMainThread {
-                            receiveEligibility(result)
-                        }
-                    }
-                } else {
-                    var convertedEligibility: [String: IntroEligibility] = [:]
-                    for (key, value) in receivedEligibility {
-                        do {
-                            let introEligibility = try IntroEligibility(eligibilityStatusCode: value)
-                            convertedEligibility[key] = introEligibility
-                        } catch {
-                            Logger.error(String(format: "Unable to create an RCIntroEligibility: %@",
-                                                error.localizedDescription))
-                        }
-                    }
-                    self.operationDispatcher.dispatchOnMainThread {
-                        receiveEligibility(convertedEligibility)
-                    }
-                }
-            }
-        // swiftlint:enable line_length
+    /**
+     * Use this function to open the manage subscriptions modal.
+     * If the manage subscriptions modal can't be opened, the managementURL in the customerInfo will be opened.
```

Comment:
Wondering whether we want to add more info on SK2 vs SK1 here? Or do we prefer to keep that black-boxed from the customer?

Reply:
I think at the end of the day the most interesting bit is what the actual behavior will be depending on the OS version, but whether that's done through SK1 or SK2 seems like an implementation detail. 
It might not even be SK at all if it's an android subscription, for example.